ingest.py

```diff
@@ -2,37 +2,36 @@
 import os
 import shutil
 import sys
+from pathlib import Path
 
-from dotenv import load_dotenv
 from langchain.docstore.document import Document
-from langchain.document_loaders import CSVLoader, PDFMinerLoader, TextLoader, UnstructuredEPubLoader, UnstructuredHTMLLoader
+from langchain.document_loaders import CSVLoader, PDFMinerLoader, TextLoader, UnstructuredEPubLoader, \
+    UnstructuredHTMLLoader, Docx2txtLoader, UnstructuredPowerPointLoader
 from langchain.embeddings import LlamaCppEmbeddings
 from langchain.text_splitter import RecursiveCharacterTextSplitter
 from langchain.vectorstores import Qdrant
 
-load_dotenv()
-llama_embeddings_model = os.environ.get("LLAMA_EMBEDDINGS_MODEL")
-persist_directory = os.environ.get("PERSIST_DIRECTORY")
-documents_directory = os.environ.get("DOCUMENTS_DIRECTORY")
-model_n_ctx = os.environ.get("MODEL_N_CTX")
+from load_env import persist_directory, chunk_size, chunk_overlap, llama_embeddings_model, model_n_ctx, \
+    documents_directory
 
+file_loaders = {  # extension -> loader
+    "txt": lambda path: TextLoader(path, encoding="utf8"),
+    "pdf": PDFMinerLoader,
+    "csv": CSVLoader,
+    "epub": UnstructuredEPubLoader,
+    "html": UnstructuredHTMLLoader,
+    "docx": Docx2txtLoader,
+    "pptx": UnstructuredPowerPointLoader,
+}
 
-def load_one_doc(filepath: str) -> list[Document]:
+
+def load_one_doc(filepath: Path) -> list[Document]:
     """load one document"""
-    if filepath.endswith(".txt"):
-        loader = TextLoader(filepath, encoding="utf8")
-    elif filepath.endswith(".pdf"):
-        loader = PDFMinerLoader(filepath)
-    elif filepath.endswith(".csv"):
-        loader = CSVLoader(filepath)
-    elif filepath.endswith(".epub"):
-        loader = UnstructuredEPubLoader(filepath)
-    elif filepath.endswith(".html"):
-        loader = UnstructuredHTMLLoader(filepath)
-    else:
-        raise ValueError(f"Unhandled file format: .{filepath.split('.')[-1]} in {filepath}")
+    if filepath.suffix[1:] not in file_loaders:
+        print(f"Unhandled file format: {filepath.name} in {filepath.parent}")
+        return []
 
-    return loader.load()
+    return file_loaders[filepath.suffix[1:]](str(filepath)).load()
```

Comment:
Nice, thanks A LOT for taking the time to tidy up the codebase. 

Although we should keep in mind readability. This pick is ok but would not further abstract it when there's no increase in performance.